components/script/dom/mutationobserver.rs

```diff
@@ -4,19 +4,42 @@
 
 use dom::bindings::codegen::Bindings::MutationObserverBinding;
 use dom::bindings::codegen::Bindings::MutationObserverBinding::MutationCallback;
-use dom::bindings::error::Fallible;
+use dom::bindings::codegen::Bindings::MutationObserverBinding::MutationObserverBinding::MutationObserverMethods;
+use dom::bindings::codegen::Bindings::MutationObserverBinding::MutationObserverInit;
+use dom::bindings::codegen::Bindings::MutationRecordBinding::MutationRecordBinding::MutationRecordMethods;
+use dom::bindings::error::{Error, Fallible};
 use dom::bindings::js::Root;
 use dom::bindings::reflector::{Reflector, reflect_dom_object};
+use dom::bindings::str::DOMString;
+use dom::mutationrecord::MutationRecord;
+use dom::node::Node;
 use dom::window::Window;
 use dom_struct::dom_struct;
+use html5ever_atoms::Namespace;
+use microtask::EnqueuedMutationCallback;
+use microtask::Microtask;
+use msg::constellation_msg::PipelineId;
+use msg::constellation_msg::PipelineIndex;
+use msg::constellation_msg::PipelineNamespaceId;
 use script_thread::ScriptThread;
+use servo_atoms::Atom;
+use std::cell::Cell;
+use std::option::Option;
 use std::rc::Rc;
 
 #[dom_struct]
 pub struct MutationObserver {
     reflector_: Reflector,
     #[ignore_heap_size_of = "can't measure Rc values"]
     callback: Rc<MutationCallback>,
+    record_queue: Vec<Root<MutationRecord>>,
+    #[ignore_heap_size_of = "can't measure DomRefCell values"]
+    options: Cell<MutationObserverInit>,
```

Comment:
Rather than storing an instance of this, store individual fields that correspond to the members of MutationObserverInit that matter - `want_attributes: Cell<bool>`, `want_attribute_old_value: Cell<bool>`, and `attribute_filter: DOMRefCell<Vec<DOMString>>`.